File: JavaPop/app/src/main/java/com/example/javapop/Models/Repository.java
package com.example.javapop.Models;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class Repository implements Parcelable {

    @SerializedName("name")
    private String nameRepository;

    @SerializedName("description")
    private String descriptionRepository;

    @SerializedName("owner")
    private OwnerGit ownerGit;

    @SerializedName("full_name")
    private String lastName;

    @SerializedName("stargazers_count")
    private int starsNumber;

    @SerializedName("forks_count")
    private int forkNumber;

    public OwnerGit getOwnerGit() {
        return ownerGit;
    }

    public String getLastName() {
        return lastName;
    }

    public String getNameRepository() {
        return nameRepository;
    }

    public String getDescriptionRepository() {
        return descriptionRepository;
    }

    public int getStarsNumber() {
        return starsNumber;
    }

    public int getForkNumber() {
        return forkNumber;
    }

    protected Repository(Parcel in) {
        this.ownerGit = in.readParcelable(OwnerGit.class.getClassLoader());
        nameRepository = in.readString();
        descriptionRepository = in.readString();
        starsNumber = in.readInt();
        forkNumber = in.readInt();
    }


    @Override
    public String toString() {
        return "Repository{" +
                "nameRepository='" + nameRepository + '\'' +
                ", descriptionRepository='" + descriptionRepository + '\'' +
                ", ownerGit=" + ownerGit +
                ", lastName='" + lastName + '\'' +
                ", starsNumber=" + starsNumber +
                ", forkNumber=" + forkNumber +
                '}';
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(ownerGit, flags);
        dest.writeString(nameRepository);
        dest.writeString(descriptionRepository);
        dest.writeInt(starsNumber);
        dest.writeInt(forkNumber);
    }

    public static final Creator<Repository> CREATOR = new Creator<Repository>() {
        @Override
        public Repository createFromParcel(Parcel in) {
            return new Repository(in);
        }

        @Override
        public Repository[] newArray(int size) {
            return new Repository[size];
        }
    };

}

File: app/src/main/java/com/uharris/desafio_android/domain/models/Repository.kt
package com.uharris.desafio_android.domain.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class Repository(
    val id: Int = 0,
    val name: String = "",
    val description: String = "",
    val owner: User = User(),
    val forks: Int = 0,
    @SerializedName("stargazers_count") val stars: Int = 0
) : Parcelable {
    constructor(source: Parcel) : this(
        source.readInt(),
        source.readString() ?: "",
        source.readString() ?: "",
        source.readParcelable<User>(User::class.java.classLoader) ?: User(),
        source.readInt(),
        source.readInt()
    )

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) = with(dest) {
        writeInt(id)
        writeString(name)
        writeString(description)
        writeParcelable(owner, 0)
        writeInt(forks)
        writeInt(stars)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Repository> = object : Parcelable.Creator<Repository> {
            override fun createFromParcel(source: Parcel): Repository = Repository(source)
            override fun newArray(size: Int): Array<Repository?> = arrayOfNulls(size)
        }
    }
}